email_processing.py

```diff
@@ -0,0 +1,197 @@
+        try:
+            print(f"Sending email with an attachment {FILE_NAME} to {EMAIL}")
+            with smtplib.SMTP_SSL('smtp.gmail.com', 465) as smtp:
+                smtp.login(SENDER_EMAIL_ADDRESS, SENDER_EMAIL_PASSWORD)
+                smtp.send_message(msg)
+                print(f'Message has been succesfully delivered to.')
+
+        except smtplib.SMTPAuthenticationError:
+            print('Username and Password not accepted.')
+
+    def list_all_documents(self):
+        """ This function retreives all documents from requesting endpoint and writes it to list as class attribute.
+        """
+        access_token = self.access_token
+        all_files = self.all_files
+        documents_endpoint = f'https://{self.API_URL}/v1/documents'
+        headers = {
+            "Authorization": f"token {access_token}"
+        }
+        try:
+            r = requests.get(documents_endpoint, headers=headers)
+            if r.status_code in range(200, 299):
+                # Pagination count to check if there is more then 1 page otherwise file won't be found
+                pagination_count = r.json()['pagination']['total_pages']
+                # I have setup 30 seconds because when I tested it less than 30 seconds was not enoght for adding a file and file was not shown in all documents
+                print('Retreiving information about all documents in all queues ... ')
+                time.sleep(30)
+                # script checks count of pages and goes page by page and concatenate results to list
+                for page in range(1, pagination_count+1):
+                    documents_endpoint = f'https://{self.API_URL}/v1/documents?page={page}'
+                    r = requests.get(documents_endpoint, headers=headers)
+                    if r.status_code in range(200, 299):
+                        all_files += r.json()['results']
+                self.all_files = all_files
+                return self.all_files
+
+        except requests.exceptions.RequestException:
+            raise Exception(f'Failed to connect to {documents_endpoint}') from None
+
+    def check_processing(self):
+        """ This function checks for latest uploaded file by arrived_at date and by filename then retreives its relevant information (annotation link)
+          If annotation link is empty -> returns The file not found and exits with code 1.
+          If annotation link is not empty -> goes to its annotation link and retreives file status by status attribute.
+        """
+        FILE_NAME = self.FILE_NAME
+        access_token = self.access_token
+        headers = {
+            "Authorization": f"token {access_token}"
+        }
+
+        # date is used for comparing and finding the lastest date
+        date = '0001-01-01T00:00:00.000Z'
+        document_list = self.all_files
+        annotation_link = None
+
+        def convertDate(d):
+            new_date = datetime.datetime.strptime(d, "%Y-%m-%dT%H:%M:%S.%fZ")
```

Comment:
just a nitpick: since python 3.7, this is possible to be done using `fromisoformat` method:
```suggestion
            new_date = datetime.datetime.fromisoformat(d)
```